### PR TITLE
Single value layout fix

### PR DESCRIPTION
**Pull request by seancurtis**
The absolute positioning pulls both the value and the placeholder out of the flow, and thus the component itself collapses down when used as a flex child. This can be solved by wrapping the component in another div, but we can solve it better by using CSS Grid instead.

before-after

### CODE REVIEW

File: packages/react-select/src/components/containers.tsx [hunk 1]
```diff
@@ -85,9 +85,10 @@ export const valueContainerCSS = <
   Group extends GroupBase<Option>
 >({
   theme: { spacing },
+  isMulti
 }: ValueContainerProps<Option, IsMulti, Group>): CSSObjectWithLabel => ({
   alignItems: 'center',
-  display: 'flex',
+  display: `${isMulti ? 'flex' : 'grid'}`,
```
Comment (Rall3n) [inline packages/react-select/src/components/containers.tsx:91]: Question of style, not critical, but why literals when you can do it without them?

**Review by Rall3n — CHANGES_REQUESTED** [review]
image

Value, placeholder and input are centered by default instead of left aligned.

**Comment by seancurtis** [comment]
@Rall3n thanks for the catch in the layout when expanded. I was working on it using a fixture to test the flex content and as such it was constantly squashed. I just reverted it before pushing. I did some more fixes to the layout/alignment as I noticed another issue too. It's much more solid now.

File: packages/react-select/src/components/Placeholder.tsx [hunk 2]
```diff
@@ -29,11 +29,9 @@ export const placeholderCSS = <
 }: PlaceholderProps<Option, IsMulti, Group>): CSSObjectWithLabel => ({
   label: 'placeholder',
   color: colors.neutral50,
+  gridArea: '1 / 2',
```
Comment (Rall3n) [inline packages/react-select/src/components/Placeholder.tsx:32]: image

Forgot to provide the `Placehholder` component with the latest fix :)

**Review by Rall3n — CHANGES_REQUESTED** [review]
One last thing, but everything else looks ok.

**Comment by seancurtis** [comment]
D'oh! Not sure how I missed that one tbh. Should be good now.

**Review by Rall3n — APPROVED** [review]
Looks good now 👍

File: packages/react-select/src/components/containers.tsx [hunk 3]
```diff
@@ -84,9 +84,10 @@ export const valueContainerCSS = <
   Group extends GroupBase<Option>
 >({
   theme: { spacing },
+  isMulti
```
Comment (Methuselah96) [inline packages/react-select/src/components/containers.tsx:87]: To make Prettier happy:
```suggestion
  isMulti,
```

**Review by JedWatson — APPROVED** [review]
Nice! Ha, love css grids, this is so neat now. Thanks @seancurtis

**Pull request merged**